ngraph/test/frontend/paddlepaddle/places.cpp

```diff
@@ -0,0 +1,474 @@
+
+/***
+model:
+
+                         [input]
+                            |
+      [const]   [const]   [transpose]
+           \      |       /
+            [ RNN (LSTM) ]
+         /      |          \
+ [transpose] [scale_{1,2}] [relu_{0,1,2}]
+     |          |              |
+  [scale_0]  [out_{1,2}]   [scale_{3,4,5}]
+     |                         |
+  [out_1]                   [out_{3,4,5}]
+
+***/
+
+std::vector<std::string> tensor_names = {
+    "x",
+    "const_1.tmp_0",
+    "const_2.tmp_0",
+    "transpose_0.tmp_0",
+    "transpose_0.tmp_1",
+    "lstm_0.tmp_0",
+    "lstm_0._generated_var_0",
+    "lstm_0.tmp_3",
+    "lstm_0.tmp_1",
+    "lstm_0.tmp_2",
+    "transpose_1.tmp_0",
+    "transpose_1.tmp_1",
+    "relu_1.tmp_0",
+    "relu_2.tmp_0",
+    "relu_3.tmp_0",
+    "save_infer_model/scale_0.tmp_1",
+    "save_infer_model/scale_1.tmp_1",
+    "save_infer_model/scale_2.tmp_1",
+    "save_infer_model/scale_3.tmp_1",
+    "save_infer_model/scale_4.tmp_1",
+    "save_infer_model/scale_5.tmp_1",
+};
+
+TEST(PDPD_Places, check_tensor_names)
+{
+    FrontEndTestUtils::setupTestEnv();
+    auto m_fem = FrontEndManager();
+    auto frontend = m_fem.load_by_framework("pdpd");
+    auto input_model = frontend->load_from_file(TEST_PDPD_MODELS + model_file);
+
+    for (const auto& tensor_name : tensor_names)
+    {
+        auto place = input_model->get_place_by_tensor_name(tensor_name);
+        EXPECT_NE(place, nullptr);
+    }
+}
+
+TEST(PDPD_Places, check_input_outputs)
+{
+    FrontEndTestUtils::setupTestEnv();
+    auto m_fem = FrontEndManager();
+    auto frontend = m_fem.load_by_framework("pdpd");
+    auto input_model = frontend->load_from_file(TEST_PDPD_MODELS + model_file);
+
+    auto inputs = input_model->get_inputs();
+    auto outputs = input_model->get_outputs();
+
+    EXPECT_EQ(inputs.size(), 1);
+    EXPECT_EQ(outputs.size(), 6);
+
+    auto tensor_place = input_model->get_place_by_tensor_name("x");
+    tensor_place->is_equal(inputs[0]);
+
+    std::vector<std::string> output_names = {"save_infer_model/scale_0.tmp_1",
+                                             "save_infer_model/scale_1.tmp_1",
+                                             "save_infer_model/scale_2.tmp_1",
+                                             "save_infer_model/scale_3.tmp_1",
+                                             "save_infer_model/scale_4.tmp_1",
+                                             "save_infer_model/scale_5.tmp_1"};
+
+    for (const auto& name : output_names)
+    {
+        const auto output_place = input_model->get_place_by_tensor_name(name);
+        auto it =
+            std::find_if(outputs.begin(), outputs.end(), [&output_place](const Place::Ptr& place) {
+                return output_place->is_equal(place);
+            });
+        EXPECT_NE(it, outputs.end());
+    }
+}
+
+// all existed in the model ops have "Out" port
+TEST(PDPD_Places, check_out_port_of_all_ops)
```

Comment:
Could these tests be moved to 'shared' folder so that some part will be common for all future frontends?

Reply:
Suggest to do it in the next PR